collection-scripts/mce-gather.sh

```diff
@@ -68,20 +68,23 @@ check_if_spoke() {
   fi
 }
 
-check_if_hypershift() {
+# parse_args parses the arguments passed to the must-gather in the form of <key>=<value>.
+# NOTE: Currently this only applies to the Hypershift must-gather.
+parse_args() {
   # get the hosted cluster name and optionally namespace
-  while [ "$1" != "" ]; do
-    FLAG=$(echo "$1" | awk -F= '{print $1}')
-    VALUE=$(echo "$1" | awk -F= '{print $2}')
-    case $FLAG in
-    hosted-cluster-name)
-      HC_NAME=$VALUE
+  while [ "${1}" != "" ]; do
+    local key value
+    key=${1%=*}
+    value=${1#*=}
+    case ${key} in
+    hosted-cluster-name) # Get the hosted cluster name
+      export HC_NAME=${value}
       ;;
-    hosted-cluster-namespace)
-      HC_NAMESPACE=$VALUE
+    hosted-cluster-namespace) # Get the hosted cluster namespace
+      export HC_NAMESPACE=${value}
       ;;
     *)
-      echo "ERROR: unknown parameter \"$FLAG\""
+      log "ERROR" "unknown parameter \"${key}\""
```

Comment:
The `log` function doesn't exist in this repo (sorry--still on my TODO to sync them up)

```suggestion
      echo "ERROR" "unknown parameter \"${key}\""
```

Reply:
ah yes, thanks